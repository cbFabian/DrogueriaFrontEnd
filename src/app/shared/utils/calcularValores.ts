export const calcularValores = (producto: string, 
    reunion: string, esFestivo: boolean) => {
        if(!producto && !reunion){
            return 0;
        }
        if(!producto){
            return JSON.parse(reunion).precio;
        }
        if(!reunion){
            return JSON.parse(producto).precio;
        }
        let valorTotal: number = JSON.parse(producto).precio + JSON.parse(reunion).precio;
        if(esFestivo){
            valorTotal *= 2;
        }                
        return valorTotal;
};
