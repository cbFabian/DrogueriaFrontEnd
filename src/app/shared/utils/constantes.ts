export const constantes = {
    HORAS24: 24,
    HORA15: 15,
    HORA19: 19,
    MINUTOS60: 60,
    SEGUNDOS60: 60,
    MILISEGUNDOS: 1000,
    POSICION_DOS: 2,
    POSICION_TRES: 3,
    POSICION_CUATRO: 4,
    NUMERO_DOS: 2,
    NUMERO_TRES: 3,
    HORAMINIMA: 4,
    NUMERO_CINCO: 5,
    HORAMAXIMA: 8,
};
