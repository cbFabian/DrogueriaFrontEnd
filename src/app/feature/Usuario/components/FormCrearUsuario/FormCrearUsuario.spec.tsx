import * as React from 'react';
import { RenderResult, fireEvent, render, wait } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';
import { FormCrearUsuario } from '.';
import { setTextEvent } from 'app/shared/utils/test';

const DOS = 2;
const TRES = 3;

describe('FormCrearUsuario test', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof FormCrearUsuario> & {
    onSubmit: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
      formTitle: 'Form test',
      onSubmit: stub(),
      irInicioSesion: stub(),
      mensajeErrorCreacion: '',
    };
    componentWrapper = render(<FormCrearUsuario {...componentProps} />);
  });

  it('should match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });

  it('should fail on submit all fields missing', async () => {
    const elem = componentWrapper.container;
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(TRES);
    expect(spans[0].textContent).toBe('El campo nombre es requerido.');
    expect(spans[1].textContent).toBe('El campo clave es requerido.');
    expect(spans[DOS].textContent).toBe('El campo de confirmar clave es requerido.');
  });

  it('should fail on submit two fields missing', async () => {
    const elem = componentWrapper.container;
    const nombre = elem.querySelector('input[name="nombre"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
      nombre && fireEvent.change(nombre, setTextEvent('nombre', 'Lorem'));
    });

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(DOS);
    expect(spans[0].textContent).toBe('El campo clave es requerido.');
    expect(spans[1].textContent).toBe('El campo de confirmar clave es requerido.');
  });

  it('should fail on submit two fields missing', async () => {
    const elem = componentWrapper.container;

    const nombre = elem.querySelector('input[name="nombre"]');
    const clave = elem.querySelector('input[name="clave"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
      nombre && fireEvent.change(nombre, setTextEvent('nombre', 'Lorem'));
    });
    await wait(() => {
      clave && fireEvent.change(clave, setTextEvent('clave', '1234'));
    });

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(1);
    expect(spans[0].textContent).toBe('El campo de confirmar clave es requerido.');
  });

  it('should submit', async () => {
    const elem = componentWrapper.container;

    const nombre = elem.querySelector('input[name="nombre"]');
    const clave = elem.querySelector('input[name="clave"]');
    const confirmarClave = elem.querySelector('input[name="confirmarClave"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
      nombre && fireEvent.change(nombre, setTextEvent('nombre', 'Lorem'));
    });
    await wait(() => {
      clave && fireEvent.change(clave, setTextEvent('clave', '1234'));
    });
    await wait(() => {
      confirmarClave && fireEvent.change(confirmarClave, setTextEvent('confirmarClave', '1234'));
    });

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });

    const formSubmitted = componentProps.onSubmit.firstCall.args[0];

    expect(formSubmitted.nombre).toBe('Lorem');
    expect(formSubmitted.clave).toBe('1234');
  });
});
