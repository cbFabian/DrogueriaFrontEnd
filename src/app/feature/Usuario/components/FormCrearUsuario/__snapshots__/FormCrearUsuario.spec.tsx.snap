// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCrearUsuario test should match snapshot 1`] = `
<div>
  <form>
    <h2>
      Form test
    </h2>
    <input
      class="sc-AxirZ eAukyj"
      name="nombre"
      placeholder="Nombre"
      value=""
    />
    <input
      class="sc-AxirZ eAukyj"
      name="clave"
      placeholder="Clave"
      value=""
    />
    <input
      class="sc-AxirZ eAukyj"
      name="confirmarClave"
      placeholder="Confirme la Clave"
      value=""
    />
    <button
      class="sc-AxjAm kgxTKy"
      type="submit"
    >
      Registrar
    </button>
  </form>
</div>
`;
